File: dofun/safe/deviceinfo/info/HardwareInfo.java
package com.dofun.safe.deviceinfo.info;

import android.content.Context;
import android.hardware.Sensor;
import android.hardware.SensorManager;
import androidx.core.util.Pair;
import com.dofun.safe.deviceinfo.utils.CommandUtils;
import java.util.ArrayList;
import java.util.List;
import org.json.JSONObject;

/* loaded from: classes8.dex */
public class HardwareInfo {
    public static JSONObject getHardwareInfoJson(Context context) {
        return CommandUtils.convertToJson(getHardwareInfo(context));
    }

    public static List<Pair<String, String>> getHardwareInfo(Context context) {
        List<Pair<String, String>> list = new ArrayList<>();
        SensorManager sensorManager = (SensorManager) context.getSystemService("sensor");
        Sensor gyroscope = sensorManager.getDefaultSensor(4);
        list.add(new Pair<>("gyroscope", gyroscope.getName()));
        Sensor magnetic = sensorManager.getDefaultSensor(2);
        list.add(new Pair<>("magnetic", magnetic.getName()));
        Sensor accelerometer = sensorManager.getDefaultSensor(1);
        list.add(new Pair<>("accelerometer", accelerometer.getName()));
        return list;
    }
}

File: app/src/main/java/com/dofun/safe/deviceinfo/info/HardwareInfo.kt
package com.dofun.safe.deviceinfo.info

import android.content.Context
import android.hardware.SensorManager
import androidx.core.util.Pair
import com.dofun.safe.deviceinfo.utils.CommandUtils
import org.json.JSONObject

object HardwareInfo {
    @JvmStatic
    fun getHardwareInfoJson(context: Context): JSONObject {
        return CommandUtils.convertToJson(getHardwareInfo(context))
    }

    @JvmStatic
    fun getHardwareInfo(context: Context): List<Pair<String, String>> {
        val list = ArrayList<Pair<String, String>>()
        val sensorManager = context.getSystemService("sensor") as SensorManager
        val gyroscope = sensorManager.getDefaultSensor(4)
        list.add(Pair("gyroscope", gyroscope.name))
        val magnetic = sensorManager.getDefaultSensor(2)
        list.add(Pair("magnetic", magnetic.name))
        val accelerometer = sensorManager.getDefaultSensor(1)
        list.add(Pair("accelerometer", accelerometer.name))
        return list
    }
}
